### PR TITLE
Add external interface messaging

### PR DESCRIPTION
**Pull request by superm1**
I filled out some of these items, maybe you can suggest some of the ones I didn't know off hand.

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [x] Documentation

### CODE REVIEW

File: plugins/acpi-dmar/README.md [hunk 1]
```diff
@@ -6,3 +6,7 @@ Introduction
 
 This plugin checks if DMA remapping for Thunderbolt devices is available. The
 result will be stored in an security attribute for HSI.
+
+External interface access
+---
```
Comment (hughsie) [inline plugins/acpi-dmar/README.md:11]: I think the markdown crazies might get upset the --- doesn't extend all the way to the `access`

File: plugins/amt/README.md [hunk 2]
```diff
@@ -25,3 +25,7 @@ Vendor ID Security
 ------------------
 
 The device is not upgradable and thus requires no vendor ID set.
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/amt/README.md:31]: /dev/mei0

File: plugins/ata/README.md [hunk 3]
```diff
@@ -47,3 +47,7 @@ This plugin uses the following plugin-specific quirks:
 |------------------------|-------------------------------------------|-----------------------|
 | `AtaTransferBlocks`    | Blocks to transfer, or `0xffff` for max   | 1.2.4                 |
 | `AtaTransferMode`      | The transfer mode, `0x3`, `0x7` or `0xe`  | 1.2.4                 |
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/ata/README.md:53]: it's an ioctl on the device node, so I'm not really sure either

File: plugins/bcm57xx/README.md [hunk 4]
```diff
@@ -26,3 +26,7 @@ Vendor ID Security
 ------------------
 
 The vendor ID is set from the PCI vendor, in this instance set to `PCI:0x14E4`
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/bcm57xx/README.md:32]: the ethtool ioctl interface on the PCI device, and in the case of the recovery device raw writes to the PCI BARs

File: plugins/coreboot/README.md [hunk 5]
```diff
@@ -57,3 +57,7 @@ Vendor ID Security
 ------------------
 
 The vendor ID is set from the BIOS vendor, in this instance `DMI:coreboot`
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/coreboot/README.md:63]: It's all DMI right now, right?

File: plugins/logind/README.md [hunk 6]
```diff
@@ -11,3 +11,7 @@ Vendor ID Security
 ------------------
 
 This protocol does not create a device and thus requires no vendor ID set.
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/logind/README.md:17]: It needs access to org.freedesktop.login1

File: plugins/msr/README.md [hunk 7]
```diff
@@ -12,3 +12,7 @@ always be disabled and locked on production hardware as it allows the
 attacker to disable other firmware protection methods.
 
 The result will be stored in a security attribute for HSI.
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/msr/README.md:18]: It needs the msr udev subsystem, and the ability to read from msr0

File: plugins/optionrom/README.md [hunk 8]
```diff
@@ -24,3 +24,7 @@ Vendor ID Security
 ------------------
 
 The device is not upgradable and thus requires no vendor ID set.
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/optionrom/README.md:30]: Read access to the rom file in PCI udev

File: plugins/tpm-eventlog/README.md [hunk 9]
```diff
@@ -15,3 +15,7 @@ Vendor ID Security
 ------------------
 
 The device is not upgradable and thus requires no vendor ID set.
+
+External interface access
+---
+???
```
Comment (hughsie) [inline plugins/tpm-eventlog/README.md:21]: Just ro /sys/kernel/security/tpm0/binary_bios_measurements I think

File: plugins/superio/README.md [hunk 10]
```diff
@@ -27,3 +27,7 @@ Vendor ID Security
 ------------------
 
 The vendor ID is set from the baseboard vendor, for example `DMI:Star Labs`
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/superio/README.md:33]: inb/outb i.e. raw reads to system memory

File: plugins/redfish/README.md [hunk 11]
```diff
@@ -73,3 +73,7 @@ and verify the uri with
 or
 
     $ curl -k https://192.168.0.133:443/redfish/v1/
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/redfish/README.md:79]: tricky; I guess http access to a given localhost URL

File: plugins/platform-integrity/README.md [hunk 12]
```diff
@@ -6,3 +6,7 @@ Introduction
 
 This plugin checks if the system SPI chip is locked. The result will be stored
 in an security attribute for HSI.
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/platform-integrity/README.md:12]: /sys/class/platform-integrity/* ro

File: plugins/pci-bcr/README.md [hunk 13]
```diff
@@ -6,3 +6,7 @@ Introduction
 
 This plugin checks if the system SPI chip is locked. The result will be stored
 in an security attribute for HSI.
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/pci-bcr/README.md:12]: Config space of random PCI devices

File: plugins/pci-mei/README.md [hunk 14]
```diff
@@ -6,3 +6,7 @@ Introduction
 
 This plugin checks if the ME is in Manufacturing Mode. The result will be stored
 in an security attribute for HSI.
+
+External interface access
+---
+????
```
Comment (hughsie) [inline plugins/pci-mei/README.md:12]: Config space of random PCI devices

Comment (superm1) [reply plugins/pci-mei/README.md:12]: But how does it reach out?  Is it some ioctl to do it?

Comment (hughsie) [reply plugins/pci-mei/README.md:12]: na, it's literally the `config` file which is exported by the kernel in sysfs -- we just read it like a normal file

Comment (superm1) [reply plugins/pci-mei/README.md:12]: OK, I think in general I want to write specifics that things need, IE `/sys/class/foo/bar/config` as that's much easier for folks to consume.

Comment (hughsie) [reply plugins/pci-mei/README.md:12]: e.g. `/sys/class/pci_bus/0000:01/device/config`

**Review by hughsie — APPROVED** [review]

**Comment by woodrow-shen** [comment]
@superm1 , Thanks for PR. It's very useful to evaluate them from core's aspect.

**Pull request merged**